pySDC/projects/PinTSimE/estimation_check.py

```diff
@@ -239,15 +242,19 @@ def plotStateFunctionAroundEvent(u_num, prob_cls_name, M_fix):  # pragma: no cov
 
                     if use_SE:
                         t_switches = [u_num[dt][M_fix][use_SE][use_A]['t_switches'] for dt in dt_list]
-                        t_switch = [t_event[i] for t_event in t_switches]
+                        for t_switch_item in t_switches:
+                            for m in range(1, len(t_switch_item)):
```

Comment:
how many iterations does this typically have? If it's a lot, you may want to speed this up using masks with sth. like
```
mask = np.append([True], np.abs(t_swtich_item[1:] - t_switch_item[:-1] > 1e-10)
t_switch_item = t_switch_item[mask]
```
If the loop is entered only a couple of times, don't bother.